pkg/cmd/cluster/cluster_unlink.go

```diff
@@ -0,0 +1,49 @@
+package cluster
+
+import (
+	"errors"
+
+	"github.com/dcos/dcos-cli/api"
+	"github.com/dcos/dcos-cli/pkg/config"
+	"github.com/spf13/cobra"
+)
+
+// newCmdClusterLink links the attached cluster to another one.
+func newCmdClusterUnlink(ctx api.Context) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:  "unlink",
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			attachedCluster, err := ctx.Cluster()
+			if err != nil {
+				return err
+			}
+
+			var linkedCluster config.Cluster
+			for _, cluster := range ctx.Clusters() {
+				if args[0] == cluster.ID() ||
+					(args[0] == cluster.Name() && ctx.IsUniqueCluster(cluster.Name())) {
+					linkedCluster = *cluster
+					break
+				}
+			}
```

Comment:
There is already a `Find` method to lookup a cluster config based on its ID / name, it is used in other cluster subcommands as well. This would remove the need for `IsUniqueCluster` in the context.